beater/handlers.go

```diff
@@ -284,76 +326,85 @@ func corsHandler(allowedOrigins []string, h http.Handler) http.Handler {
 
 			w.Header().Set("Content-Length", "0")
 
-			sendStatus(w, r, http.StatusOK, nil)
+			sendStatus(w, r, ok)
 
 		} else if validOrigin {
 			// we need to check the origin and set the ACAO header in both the OPTIONS preflight and the actual request
 			w.Header().Set("Access-Control-Allow-Origin", origin)
 			h.ServeHTTP(w, r)
 
 		} else {
-			sendStatus(w, r, http.StatusForbidden, errForbidden)
+			sendStatus(w, r, forbidden)
 		}
 	})
 }
 
 func processRequestHandler(pf ProcessorFactory, config conf.Config, report reporter, decode decoder.Decoder) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		code, err := processRequest(r, pf, config, report, decode)
-		sendStatus(w, r, code, err)
+		res := processRequest(r, pf, config, report, decode)
+		sendStatus(w, r, res)
 	})
 }
 
-func processRequest(r *http.Request, pf ProcessorFactory, config conf.Config, report reporter, decode decoder.Decoder) (int, error) {
+func processRequest(r *http.Request, pf ProcessorFactory, config conf.Config, report reporter, decode decoder.Decoder) serverResponse {
 	processor := pf()
 
 	if r.Method != "POST" {
-		return http.StatusMethodNotAllowed, errPOSTRequestOnly
+		return methodNotAllowed
 	}
 
 	data, err := decode(r)
 	if err != nil {
-		return http.StatusBadRequest, errors.Wrap(err, "while decoding")
+		if strings.Contains(err.Error(), "request body too large") {
+			return requestTooLarge
+		}
+		return cantDecode(err)
+
 	}
 
 	if err = processor.Validate(data); err != nil {
-		return http.StatusBadRequest, err
+		return cantValidate(err)
 	}
 
 	payload, err := processor.Decode(data)
 	if err != nil {
-		return http.StatusBadRequest, err
+		return cantDecode(err)
 	}
 
 	if err = report(pendingReq{payload: payload, config: config}); err != nil {
-		return http.StatusServiceUnavailable, err
+		return cantPublish(err)
```

Comment:
Why don't you distinguish between `errChannelClosed` and `errFull` error here? I think a seperate counter for a full queue would totally make sense.

Reply:
mostly to keep the code simpler and not have to check here the error message from queueFull /  channelClosed, which can change and this wouldn't work anyways.
Im fine having them together because `errchannelClosed` only happens during shutdown and metrics are sent every 30 seconds, so only 1 doc for the whole process lifetime would have an inaccurate server full count.

but i ca separate them if you prefer

Reply:
As you are splitting all the others, and the `queue_full` is one of the most common/interesting errors I'd prefer to have them separated.